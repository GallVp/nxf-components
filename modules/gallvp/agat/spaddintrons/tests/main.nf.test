nextflow_process {

    name "Test Process AGAT_SPADDINTRONS"
    script "../main.nf"
    process "AGAT_SPADDINTRONS"

    tag "modules"
    tag "modules_gallvp"
    tag "agat"
    tag "agat/spaddintrons"

    test("homo_sapiens - genome - gtf") {


        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr1/genome.gtf', checkIfExists: true)
                ]
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo_sapiens - genome - gtf - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr1/genome.gtf', checkIfExists: true)
                ]
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}