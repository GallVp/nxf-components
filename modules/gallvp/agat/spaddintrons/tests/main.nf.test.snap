{
    "homo_sapiens - genome - gtf - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b2765a6bc10652f833965c022b025370"
                ],
                "gff": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b2765a6bc10652f833965c022b025370"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-10T10:10:03.150607"
    },
    "homo_sapiens - genome - gtf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff:md5,a798830c6ad0a27b1564b98e1a1fc3aa"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b2765a6bc10652f833965c022b025370"
                ],
                "gff": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff:md5,a798830c6ad0a27b1564b98e1a1fc3aa"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b2765a6bc10652f833965c022b025370"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-10T10:09:58.155568"
    }
}